packages/ctl/ctl.js

```diff
@@ -104,6 +160,66 @@ masterConfigCommands.add(new libCommand.Command({
 		await libLink.messages.setMasterConfigProp.send(control, request);
 	},
 }));
+masterConfigCommands.add(new libCommand.Command({
+	definition: ["edit [editor]", "Edit master configuration", (yargs) => {
+		yargs.positional("editor", {
+			describe: "Editor to use",
+			type: "string",
+			default: "",
+		});
+	}],
+	handler: async function(args, control) {
+		let response = await libLink.messages.getMasterConfig.send(control);
+		let tmpFile = await libFileOps.getTempFile("ctl-", "-tmp", os.tmpdir());
+		let editor = await getEditor(args.editor);
+		if (editor === -1) {
+			throw new libErrors.CommandError(`No editor avalible. Checked CLI input, EDITOR and VISUAL env vars
+							  Try "ctl master config edit <editor of choice>"`);
+		}
+		let allConfigElements = await serializedConfigToString(respose.serialized_config, libConfig.MasterConfig, {});
+		await fs.writeFile(tmpFile, allConfigElements, (err) => {
+			if (err) {
+				throw err;
+			}
+		});
+		let editorSpawn = child_process.spawn(editor, [tmpFile], {
+			stdio: "inherit",
+  			detached: false,
+		});
+		editorSpawn.on("data", (data) => {
+  			process.stdout.pipe(data);
+		});
+		let doneEmitter = new events.EventEmitter();
+		editorSpawn.on("exit", async (exit) => {
+			const data = await fs.readFile(tmpFile, "utf8");
+			const final = await configToKeyVal(data);
+			for (let index in final) {
+				if (index in final) {
+					try {
+						await libLink.messages.setMasterConfigField.send(control, {
```

Comment:
@Hornwitser is there a way to set config options in batch mode? Having to loop over the options and set each individually like this seems like not the best way to do it :)

Reply:
I’ve had a look through lib/link and I can’t find one, but there may be one I don’t know about.